src/Mafia/Hash.hs

```diff
@@ -52,25 +54,25 @@ renderHashError = \case
 
 newtype Hash =
   Hash {
-      unHash :: ByteString
+      unHash :: Digest SHA1
     } deriving (Eq, Ord, Show)
 
 renderHash :: Hash -> Text
 renderHash =
-  T.decodeUtf8 . Base16.encode . unHash
+  T.decodeUtf8 . takeBase16 . unHash
 
 parseHash :: Text -> Maybe Hash
 parseHash hex =
-  let (bs, _) = Base16.decode (T.encodeUtf8 hex)
-  in if B.length bs == 20
-     then Just (Hash bs)
-     else Nothing
+  let
+    (bs, _) = Base16.decode (T.encodeUtf8 hex)
+  in
+    Hash <$> Hash.digestFromByteString bs
```

Comment:
It is now done by [digestFromByteString](http://hackage.haskell.org/package/cryptonite-0.15/docs/Crypto-Hash.html#v:digestFromByteString)
> Try to transform a bytearray into a Digest of specific algorithm.
> If the digest is not the right size for the algorithm specified, then Nothing is returned.